### PR TITLE
refactor(golang): Use format strings

### PR DESCRIPTION
**Pull request by jletey**
#### Description

This PR deprecates `symbol` `version` `style` keys in `GoConfig`, and adds `format` key instead.

Based off of #1063.

#### Motivation and Context

Related #1057

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):

Screenshot 2020-04-08 at 17 28 45

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by heyrict** [comment]
Oops, this should be a bug. Wait a minute and I'll add corresponding tests to it.

File: src/configs/go.rs [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
 
-use ansi_term::{Color, Style};
 use starship_module_config_derive::ModuleConfig;
 
 #[derive(Clone, ModuleConfig)]
 pub struct GoConfig<'a> {
-    pub symbol: SegmentConfig<'a>,
-    pub version: SegmentConfig<'a>,
-    pub style: Style,
+    pub format: &'a str,
```
Comment (heyrict) [inline src/configs/go.rs:7]: Symbol and style should not be deleted as is discussed in https://github.com/starship/starship/discussions/1116#discussioncomment-6579

File: src/modules/golang.rs [hunk 2]
```diff
@@ -33,15 +34,24 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
...
     let config: GoConfig = GoConfig::try_load(module.config);
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter.map(|variable| match variable {
+            "version" => Some(module_version.clone()),
+            _ => None,
+        })
+    } else {
+        log::warn!("Error parsing format string in `golang.format`");
+        return None;
+    };
```
Comment (heyrict) [inline src/modules/golang.rs:49]: The API of `StringFormatter::map` has changed. Try merging latest `feat/format-string`.

**Comment by jletey** [comment]
@heyrict Thanks for the review! I have made the suggested changes :)

The tests are not passing because of an issue with Python versions. But not to worry, I have opened #1181 to fix this.

**Pull request merged**